pre_commit/languages/rust.py

```diff
@@ -52,15 +93,57 @@ def _add_dependencies(
         f.truncate()
 
 
+def install_rust_with_toolchain(toolchain: str) -> None:
+    # Check if rustup is already present in PATH
+    rustup_exe = parse_shebang.find_executable('rustup')
+    with tempfile.TemporaryDirectory() as rustup_dir:
+        rustup_env = (
+            ('RUSTUP_HOME', rustup_dir),
+        )
+        with envcontext(rustup_env):
+            if rustup_exe is None:
+                # We did not detect rustup and need to download it first.
+                if sys.platform == 'win32':
+                    if platform.machine() == 'x86_64':
+                        url = 'https://win.rustup.rs/x86_64'
+                    else:
+                        url = 'https://win.rustup.rs/i686'
+                else:
```

Comment:
I think windows can also run on arm64 -- might need to make a mapping of architectures and platforms to select the url out of that

Reply:
I'm not a windows expert, but IIUC Windows on ARM has x86 emulation built in: https://learn.microsoft.com/en-us/windows/arm/overview#build-windows-apps-that-run-on-arm